src/indexer/allo/v2/handleEvent.ts

```diff
@@ -25,6 +25,21 @@ function generateRoundRoles(poolId: bigint) {
   return { managerRole, adminRole };
 }
 
+async function getProjectType(db: Database, profileId: Hex, metadata: JSON) {
+
+  if (metadata.hasOwnProperty("canonical")) {
+    // Linked Profile
+    const canonical: CanonicalProject = (metadata as any)["canonical"] as CanonicalProject;
+    const project = await db.getProjectById(canonical.chainId, profileId);
+    if (!project) {
+      throw new Error("Project not found");
+    }
```

Comment:
I suggest not to throw an error here. The project might not exist yet if the other chain is slower to index

Reply:
I would rather set the `type` based on the metadata without checking the db